### DISCUSSION TITLE
Make a request other than HTTP request

### DISCUSSION THREAD

**Discussion by mschoettle** [unpoly/unpoly#512] [Q&A]
We have a mobile app that makes requests through some kind of message queue. The app is basically a web app with a wrapper around to make it mobile. I was wondering if we could build the/a new UI in our backend (Django) and use unpoly in the app to make requests, through this queue instead of a good old HTTP request.  Receiving HTML instead of JSON and swapping out fragments would be much easier to handle.

Is it generally possible to change how unpoly makes a request?

**Comment by triskweline** [comment]
Unpoly only talks HTTP.

Maybe you could implement a service worker that intercepts HTTP calls made by Unpoly and tunnels them through your message queue.

We also maintain a web app that has a mobile wrapper, but that one just does HTTP requests.
